rules/active_ruleset.go

```diff
@@ -744,8 +746,36 @@ func (res *ruleMatchResults) unique() *ruleMatchResults {
 	if len(res.pipelines) == 0 {
 		return res
 	}
+
+	// First resolve any drop policies
```

Comment:
So the reason `ReverseMatch` is not needed is because basically you don't forward the metric at all if forward matching resolves in "drop".  So that way it would never make it to storage or the query service if "drop" is applied.

Reply:
I understand that we don't need to forward data between (t1,t2), but there could be data between (t0,t1) and (t2,t3) for a metric. I think it's better to have the `ReverseMatch` logic here to return the correct matched result for an ID given all the rules, otherwise the users of `ReverseMatch` call (like query service) will have to resolve the storage policies between (t1, t2) and even handle the logic of `DROP_MUST` and `DROP_IF_ONLY_MATCH` to figure out that data is being dropped (t1, t2).

Reply:
So the query service doesn't need to handle drop logic at all, it should ignore it.  

Let me investigate exact call sites but essentially this should only effect emission of metrics, not aggregation nor queries of them.